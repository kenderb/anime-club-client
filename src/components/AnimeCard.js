import React from 'react';
import PropType from 'prop-types';
import { connect } from 'react-redux';
import './AnimeCard.style.css';
import { Link } from 'react-router-dom';

const AnimeCard = ({ anime }) => {
  const { title, url_image: imageUrl, id } = anime;

  return (
    <li className="anime-card">
      <img src={imageUrl} alt={title} className="anime-image" />
      <p>{title}</p>
      <Link to={`/animes/${id}`}> See more </Link>
    </li>
  );
};
AnimeCard.propTypes = {
  anime: PropType.instanceOf(Object).isRequired,
};

const mapStateToprops = state => ({
  user: state.user,
  favorites: state.favorites,
});

export default connect(mapStateToprops)(AnimeCard);
