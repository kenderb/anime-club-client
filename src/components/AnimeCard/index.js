import React from 'react';
import PropType from 'prop-types';
import { connect } from 'react-redux';
import './AnimeCard.style.css';
import { Link, useHistory } from 'react-router-dom';

const AnimeCard = ({ anime, onClickImage, previusPath }) => {
  const history = useHistory();
  const { title, url_image: imageUrl, id } = anime;
  const currentPath = previusPath === '/dashboard' ? 'Dashboard' : 'FavoriteList';

  return (
    <li className="anime-card" onClick={() => onClickImage(anime.id)} role="presentation">
      <img src={imageUrl} alt={title} className="anime-image" />
      <p>
        {title}
        <br />
        <Link to={`/animes/${id}`} onClick={() => history.push(`/animes/${id}`, { from: currentPath })}> See more </Link>
      </p>
    </li>
  );
};
AnimeCard.propTypes = {
  anime: PropType.instanceOf(Object).isRequired,
  previusPath: PropType.string,
  onClickImage: PropType.func,
};

AnimeCard.defaultProps = {
  onClickImage: () => {},
  previusPath: '',
};

const mapStateToprops = state => ({
  user: state.user,
  favorites: state.favorites,
});

export default connect(mapStateToprops)(AnimeCard);
