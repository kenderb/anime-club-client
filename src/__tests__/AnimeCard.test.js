import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { createStore, applyMiddleware } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import combineReducers from '../reducers';
import AnimeCard from '../components/AnimeCard';

test('AnimeCard renders correctly', () => {
  const mockObj = { title: 'some title' };
  const store = createStore(combineReducers, applyMiddleware(thunk));
  const tree = create(
    <Provider store={store}>
      <BrowserRouter>
        <AnimeCard anime={mockObj} />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
