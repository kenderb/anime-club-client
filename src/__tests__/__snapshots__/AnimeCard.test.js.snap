// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeCard renders correctly 1`] = `
<li
  className="anime-card"
  onClick={[Function]}
  role="presentation"
>
  <img
    alt="some title"
    className="anime-image"
  />
  <p>
    some title
    <br />
    <a
      href="/animes/undefined"
      onClick={[Function]}
    >
       See more 
    </a>
  </p>
</li>
`;
